### DISCUSSION TITLE
Configuring an exiting language server

### DISCUSSION THREAD

**Discussion by the42** [joaotavora/eglot#1109] [Q&A]
I read https://joaotavora.github.io/eglot/#Setting-Up-LSP-Servers From there I know in priciple how to setup a new language server.

I would want to change the way the Eglot-Integration of [TypeScript Language Server](https://github.com/typescript-language-server/typescript-language-server) formats my javascript-code.

Typescript LS supports these options to change formatting:
https://github.com/typescript-language-server/typescript-language-server#workspacedidchangeconfiguration

I am interested in setting the there described `[language].format.placeOpenBraceOnNewLineForControlBlocks: boolean;` to true.

* How can I do this? Do I have to re-create the settings for https://github.com/joaotavora/eglot/blob/master/eglot.el#L189 ? 
* What would the configuration look like, when I add `:initializationOptions options…` to this setting? Forgive me here for this simple question but ELisp is more or less unknown to me and I couldn't find examples.
* Supposed there would be already a pre-defined `:initializationOptions options…` in eglot.el:  Is there a way to add to default `:initializationOptions` (eg. in `inti.el`) without re-creating the complete definition for a specific eglot LS configuration?

**Comment by joaotavora** [comment]
Read the section about `eglot-workspace-configuration` in  https://joaotavora.github.io/eglot/#Customizing-Eglot.

This used to be its own separate section, I will try to make it so again.
